packages/kotti-ui/source/kotti-field-inline-edit/KtFieldInlineEdit.vue

```diff
@@ -0,0 +1,161 @@
+<template>
+	<div class="kt-field-inline-edit">
+		<div class="kt-field-inline-edit__input">
+			<KtFieldTextArea
+				v-if="isMultiLine"
+				v-bind="multiLineProps"
+				:disabled="!isEditing"
+				@blur="handleBlur"
+				@focus="handleFocus"
+				@input="handleInput"
+			/>
+			<KtFieldText
+				v-else
+				v-bind="singleLineProps"
+				:disabled="!isEditing"
+				@blur="handleBlur"
+				@focus="handleFocus"
+				@input="handleInput"
+			/>
+		</div>
+		<div v-if="isEditing" class="icon yoco" v-text="Yoco.Icon.CHECK" />
+	</div>
+</template>
+
+<script lang="ts">
+import { Yoco } from '@3yourmind/yoco'
+import { defineComponent, computed } from '@vue/composition-api'
+
+import { makeProps } from '../make-props'
+import { Kotti } from '../types'
+
+import { KottiFieldInlineEdit, Shared } from './types'
+
+export default defineComponent<KottiFieldInlineEdit.PropsInternal<Shared.Mode>>(
```

Comment:
Shouldn’t this generic be moved to `setup(props)`?

Reply:
well you technically can't have that if you use makeProps

Reply:
Ah, fair enough I guess if that’s a block right now